azuredevops/internal/service/build/resource_resource_authorization.go

```diff
@@ -89,6 +90,10 @@ func resourceResourceAuthorizationRead(d *schema.ResourceData, m interface{}) er
 		})
 
 		if err != nil {
+			if utils.ResponseWasNotFound(err) {
```

Comment:
@xuzhang3 @logachev Do you think adding an integration test would be worth it to showcase this issue?

Reply:
This exists in all resources, so I think it worth having at least a unit test that covers this, but it should be resource agnostic.. (Or an easy way to run it for a specific resource..) 

So, 100% agreed unit test would help catch every occasion we don't check for Not Found, but it is out of scope for the PR..